### DISCUSSION TITLE
关于解除U盘限制的思考

### DISCUSSION THREAD

**Discussion by Kifranei** [ZiHaoSaMa66/OsEasy-ToolBox#47] [General]
我们学校也用了噢易云，经过今天实测发现闪迪cz880的非mlc版本（115GB）的，插电脑显示本地磁盘的U盘可以不受禁用U盘限制正常使用U盘，其他插入显示为U盘的U盘会受到限制，可不可以把显示为U盘的U盘模拟为本地磁盘来规避限制？

**Comment by ZiHaoSaMa66** [comment]
尝试一下在插入被限制读取的U盘 用`ToatlCommander`或者`xyplorer`这种第三方的资源管理器来读取看看能否正常读取文件

**Comment by ZiHaoSaMa66** [comment]
U盘模拟为本地磁盘这个方向我也去尝试下

**Comment by SheepYhangCN** [comment]
在教师端可以选择三个开关
禁用U盘 禁用外部磁盘 禁用USB外接设备
没有被限制只是因为教师端没有开
如果教师端那边想 任何USB设备都可以被禁掉
所以这个方法大抵是不行

**Comment by ZiHaoSaMa66** [comment]
难绷 那想要搞定这个只能用注入之类的方案了
可惜主播不会搞啊（捂脸

**Comment by Kifranei** [comment]
实测USB禁用前插入固态U盘，禁用后有几次是可以正常访问，但如果禁用后从没插过 插入会显示拒绝访问。此问题或许需要逆向噢易云控制程序来解决

**Comment by SheepYhangCN** [comment]
我在想能不能通过类似目前拦截投屏的方式拦截到USB和网络锁的命令
然后模拟命令发出解禁信号

如果没办法模拟出解禁命令，目前是没有办法在被锁定的情况下删除限制的

#31

**Comment by ZiHaoSaMa66** [comment]
> 我在想能不能通过类似目前拦截投屏的方式拦截到USB和网络锁的命令 然后模拟命令发出解禁信号
> 
> 如果没办法模拟出解禁命令，目前是没有办法在被锁定的情况下删除限制的
> 
> #31

在老版本`v10.5`的时候是可以用类似的方式实现解控 现在这个方式似乎失效了
目前思路是 
1. 直接调用DLL内的函数来实现解限 

2. 再者是替换老的`x64/DeviceControl_x64.exe` 并且用如下命令解除

`DeviceControl_x64.exe --type net --operation 0`
`DeviceControl_x64.exe --type usb --operation 0`
[Lite ToolBox Code](https://github.com/ZiHaoSaMa66/OsEasy-ToolBox-Lite/blob/main/src/remain.py)

**Comment by ZiHaoSaMa66** [comment]
上机测试了下 Usb的测试用例几乎都不可用
返回值为0 也没有将u盘进行锁定或解除

```
file name: easyusbctrl.dll
function name: EasyUsb_StartWorking
return type: int

[DEBUG] dll result: 0
```

```
file name: easyusbctrl.dll
function name: EasyUsb_StopWorking
return type: int

[DEBUG] dll result: 0
```

反而解除和恢复网络控制的测试用例能用

```
file name: OeNetlimit.dll
function name: DisableInternet
return type: int

SetSpeedControl ok
[DEBUG] dll result: -2147133183
[DEBUG] Error message:
```

```
file name: OeNetlimit.dll
function name: EnableNet
return type: int

SetSpeedControl ok
[DEBUG] dll result: -2147046399
[DEBUG] Error message:
```

测试时学生端版本`v10.9.0.4881`
